### ISSUE TITLE
Please update the doc to indicate that glmatrix needs to be included for the examples to work

### ISSUE THREAD

**Issue by Vaporbook** [mdn/webgl-examples#20]
Hi,

The doc at https://developer.mozilla.org/en-US/docs/Web/API/WebGL_API/Tutorial/Adding_2D_content_to_a_WebGL_context doesn't mention the dependency on glmatrix as an external script (except for one oblique reference to it in a comment that I saw after debugging extensively and googling about this).

As it stands, most folks will run into the issue of mat4 being undefined, which is confusing in particular because it's a type in the gl scripts. I added a script src to https://mdn.github.io/webgl-examples/tutorial/gl-matrix.js in order to resolve the dependency. I think the doc just needs to mention it more explicitly as a basic requirement for the tutorial.

Thanks,
Aaron

**Comment by shawnwhinnery** [comment]
May I suggest you remove the dist  and use the glmatrix source code instead.

The instructions depend on the reader being able to decompile a web pack bundle in order to understand the lesson.

**Comment by TomasHubelbauer** [comment]
I have added GLMatrix using JSDelivr like this:

```html
<script src="https://cdn.jsdelivr.net/npm/gl-matrix@3.0.0/gl-matrix-min.js"></script>
```

The note in the MDN article which takes one to this issue should be _before_ the `mat4` object is used for the first time, not after!

**Comment by TomasHubelbauer** [comment]
Yeah I guess the `gl-matris` library has had a breaking change since and `mat4` is no longer a global it sets on `window` (maybe it is now only accessible with a bundler). I just ended up changing the script source to `https://mdn.github.io/webgl-examples/tutorial/gl-matrix.js` in my personal projects where I am not using a bundler. This URL is from the MDN article and I assume it is a copy-pasted older version of `gl-matrix` they pasted into the repository. It is accessible because the URL is for GitHub Pages not the GitHub repository so there are no CORS/MIME issues or stuff like that.

**Comment by brianzhou13** [comment]
Thanks @TomasHubelbauer -- replacing the `jsdelivr` source with the `mdn.github.io` source worked for me.

**Comment by alanvgreen** [comment]
None of these quite worked for me. I ended up adding
    <script src="https://cdn.jsdelivr.net/npm/gl-matrix@3/gl-matrix.js"></script>
inside my <head> tag.

In the JS I then added
    const mat4 = glMatrix.mat4.

This got example 2 going.
